### ISSUE TITLE
Bug in Config layer

### ISSUE THREAD

**Issue by tclune** [esmf-org/esmf#192]
[reproducer_192.tgz](https://github.com/esmf-org/esmf/files/13925211/reproducer_192.tgz)

The following program produces incorrect results when run using the resource file below.
```f90
program main
  use ESMF
  implicit none
  type(ESMF_VM) :: vm
  type (ESMF_Config) :: cf
  character*100 :: fname
  logical :: ispresent
  character(len=ESMF_MAXSTR) :: string
  integer :: status

  call ESMF_Initialize(vm=vm, rc=status)

  fname = 'test_config.rc'
  cf = ESMF_ConfigCreate(rc=status)
  call ESMF_ConfigLoadFile( cf, fname, unique = .true., rc=status)


  call ESMF_ConfigFindLabel(cf, 'CB',  isPresent=ispresent, rc=status)
  print*, '----'
  print*, 'bf configset-A,  CB: ispresent=', ispresent

  string='A1XXX'
  call ESMF_ConfigSetAttribute( cf, trim(string), label='CA:', rc=status)

  call ESMF_ConfigFindLabel(cf, 'CB',  isPresent=ispresent, rc=status)
  print*, '----'
  print*, 'af configset-A,  CB: ispresent=', ispresent

end program main
```
`test_config.rc:`:
```
CA: 'A1'
CB: 'B1'
```

**Comment by tclune** [comment]
Found output:
```
 /Users/tclune/Fortran/ESMF_config_bug/main.F90          19 T           0
 /Users/tclune/Fortran/ESMF_config_bug/main.F90          23           0
 /Users/tclune/Fortran/ESMF_config_bug/main.F90          26 F           0
```


A clue is that the reproducer depends on the new value being a longer string than the original from the rc file.

**Comment by theurich** [comment]
Likely not compiler dependent, but just in case, Tom says that he ran the reproducer on NAG and gfortran-13 to see it fail.

**Comment by tclune** [comment]
To clarify:  I've only had the failure with gfortran 12.3.  NAG succeeds.  @mathomp4 has been unable to replicate on his nigh identical laptop.   We are mystified.

**Comment by tclune** [comment]
Sorry I confused this ticket with my open issue on "Info" (not Config).

**Comment by tclune** [comment]
Understood.   Really too bad that it cannot be just jetisoned in favor of YAML.    Backward compatibility and all that. :-)

**Comment by danrosen25** [comment]
> Understood. Really too bad that it cannot be just jetisoned in favor of YAML. Backward compatibility and all that. :-)

Agreed, but I know that there are A LOT of workflows out there using ESMF's proprietary config. We probably won't be able to abandon ESMF's proprietary config format in v9.0.0 either but that's a discussion for the CRB.

**Comment by danrosen25** [comment]
This fix will be introduced in v8.6.1
